TYPO3.Flow/Classes/TYPO3/Flow/Http/Component/TrustedProxiesComponent.php

```diff
@@ -52,11 +52,20 @@ public function handle(ComponentContext $componentContext)
         $hostHeader = $this->getFirstTrustedProxyHeaderValue(self::HEADER_HOST, $trustedRequest);
         $portFromHost = null;
         if ($hostHeader !== null) {
-            $portSeparatorIndex = strrpos($hostHeader, ':');
-            if ($portSeparatorIndex !== false) {
-                $portFromHost = substr($hostHeader, $portSeparatorIndex + 1);
-                $trustedRequest->getUri()->setPort($portFromHost);
-                $hostHeader = substr($hostHeader, 0, $portSeparatorIndex);
+            if (strpos($hostHeader, '[') === 0 && strrpos($hostHeader, ']') !== false) {
+                $portSeparatorIndex = strrpos($hostHeader, ':', -strrpos($hostHeader, ']'));
+                if ($portSeparatorIndex !== false) {
```

Comment:
This block could be moved outside of the brackets check to calculate the `$portSeparatorIndex`.